pkg/auth/rights/rightsutil/rightsutil.go

```diff
@@ -22,16 +22,13 @@ import (
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
 )
 
-// EventIsVisible returns whether evt is visible given rights in the context.
-func EventIsVisible(ctx context.Context, evt events.Event) (bool, error) {
-	visibility := evt.Visibility().Union(ttnpb.RightsFrom(
-		ttnpb.RIGHT_APPLICATION_ALL,
-		ttnpb.RIGHT_CLIENT_ALL,
-		ttnpb.RIGHT_GATEWAY_ALL,
-		ttnpb.RIGHT_ORGANIZATION_ALL,
-		ttnpb.RIGHT_USER_ALL,
-	))
-	for _, entityIDs := range evt.Identifiers() {
+// EventIsVisible returns whether ev is visible given rights in the context.
+func EventIsVisible(ctx context.Context, ev events.Event) (bool, error) {
+	visibility := ev.Visibility().Implied()
```

Comment:
This is not correct in this context. 

Maybe `Implied()` is a bit confusing. If a user has the "admin right", that implies that the user has all rights. If an event is visible to callers with the "admin right", that doesn't mean it's also visible to the implicit rights (i.e. for a caller with only the "info right").